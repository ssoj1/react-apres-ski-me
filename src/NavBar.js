import React from "react";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";
import "./NavBar.css";

/** Navigation bar for sites. Shows up on every page. 
 * 
 * Props: 
 * - none
 * 
 * State: 
 * - none 
 * 
 * App -> NavBar
*/

function NavBar() {
  console.debug("* NavBar");

  return (
    <div>
      <Navbar expand="md">
        <NavLink exact to="/" className="navbar-brand">
        Apr√®s-Ski-Me
        </NavLink>

        <Nav className="ml-auto" navbar>
          <NavItem>
            <NavLink exact to="/snacks">Snacks</NavLink>
          </NavItem>

          <NavItem>
            <NavLink exact to="/drinks">Drinks</NavLink>
          </NavItem>

          <NavItem>
            <NavLink exact to="/additem">Add An Item</NavLink>
          </NavItem>

        </Nav>
      </Navbar>
    </div>
  );
}

export default NavBar;
