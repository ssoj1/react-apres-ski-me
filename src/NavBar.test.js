import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "./NavBar";

test("renders without crashing", function () {
  render(
      <MemoryRouter>
          <NavBar />
      </MemoryRouter>,
  );
});

test("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
          <NavBar />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

