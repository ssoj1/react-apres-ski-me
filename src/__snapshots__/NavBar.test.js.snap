// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Apr√®s-Ski-Me
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            href="/snacks"
          >
            Snacks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/drinks"
          >
            Drinks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/additem"
          >
            Add An Item
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
